### ISSUE TITLE
1. For transactions which have a {`null`, `void`, `none`, `empty`, `missing`, category ID `-1`} category value, please change the category value from `Unknown` to `Uncategorized` (en-US locale) or `Uncategorised` (en-UK, en-AU, en-NZ locales); 2. Please allow translations.

### ISSUE THREAD

**Issue by ovari** [moneymanagerex/moneymanagerex#5924]
1. For transactions which have a {`null`, `void`, `none`, `empty`, `missing`, category ID `-1`} category value, please change the category value from `Unknown` to `Uncategorized` (en-US locale) or `Uncategorised` (en-UK, en-AU, en-NZ locales).
2. Please allow the default text `Uncategorized` (en-US locale) or `Uncategorised` (en-UK, en-AU, en-NZ locales) to be translatable.

* https://github.com/moneymanagerex/moneymanagerex/issues/5904#issuecomment-1549227219
* https://github.com/moneymanagerex/moneymanagerex/issues/5904#issuecomment-1549533197

What do you think?

Thank you

**Comment by ovari** [comment]
`Uncategorized` could also be written as `Not Categorized` or `Not categorized`

**Comment by ovari** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/5583#issuecomment-1407462396

**Comment by ovari** [comment]
1. If a transaction has no category, should it be shown with void/nothing/blank (not sure the correct word to use)?
2. If a report is generated by category, should transactions without categories have the category heading `Uncategorized`?
3. It would nice if there was an option to select all transactions which have no category. This would enable these transactions to be more easily found and then categories assigned to them.

Thank you

**Comment by tactilis** [comment]
@ovari 
> 1. If a transaction has no category, should it be shown with void/nothing/blank (not sure the correct word to use)?

No, it should not.

Firstly, a transaction cannot have "no category". The category might be "Unknown" if the transaction it was imported from a CSV file where the category data was blank - but Category is a mandatory field and it was always have a value in MMEX.

Secondly, the Status of a transaction (U/R/V/F/D) is a user configurable value that conveys information that has nothing to to do with the categorisation of the transaction.

<br>

> 2. If a report is generated by category, should transactions without categories have the category heading Uncategorized?

We already have a category name for this; i.e. 'Unknown'. I don't see any real benefit from changing this.

<br>

> 3.  It would nice if there was an option to select all transactions which have no category. This would enable these transactions to be more easily found and then categories assigned to them.

What do you mean by "no category"? Have you somehow got transactions in your database which have an empty category field? If so, this is an error. Do you know how they arose?

**Comment by ovari** [comment]
> Category is a mandatory field and it was always have a value in MMEX.

Why? Would this break importing from other programs?

> The category might be "Unknown" if the transaction it was imported from a CSV file

Intuit/Quicken also allow entering transactions without a Category.

> What do you mean by "no category"? Have you somehow got transactions in your database which have an empty category field? If so, this is an error. Do you know how they arose?

A user might enter transactions with a payee and amount with a category added at a later time.

**Comment by tactilis** [comment]
> A user might enter transactions with a payee and amount with a category added at a later time.

Have you even tried this?

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/6fd4ae5b-86cf-4ca9-a941-08ebeaae8cab)

It's a mandatory field.

<br>

Sorry, but it feels that you keep posing problems that are not actually real issues that you are experiencing.

**Comment by vomikan** [comment]
I've been thinking... There may be an interesting feature that if transactions are in the FollowUp status, then the category (and payee) can not be filled in.

**Comment by ovari** [comment]
> I've been thinking... There may be an interesting feature that if transactions are in the FollowUp status, then the category (and payee) can not be filled in.

If there are transactions imported (from CSV or QIF formats) and the category is blank, should MMEX also set the status as FollowUp?

If a transaction is imported from QIF and the transaction is Reconciled, there may be a problem if FollowUp and Reconciled can both not be selected in MMEX.

**Comment by vomikan** [comment]
The fact is that when importing, if there is no category, then one with the name (Somthing like) "No caterogy" is created.

There is no option not to set a category.

And I'm talking about making such an opportunity.

**Comment by tactilis** [comment]
@vomikan
> The fact is that when importing, if there is no category, then one with the name (Somthing like) "No caterogy" is created.
There is no option not to set a category.

The `Unknown` category already gets applied.

If you allow the option not to set a category, then instead of having a slice on the report pie charts that says `Unknown`, then we'd need to have a slice that says 'Blank' or 'Empty' or 'Not set' or 'No category'.

You're just swapping one word `Unknown` for another.

I don't understand the point of this.


@ovari
> If there are transactions imported (from CSV or QIF formats) and the category is blank, should MMEX also set the status as FollowUp?

I don't agree with this. At present, Follow Up is purely under the user's control. I set Follow Up on certain transactions for a specific reason. I don't want Import to potentially add a load of Follow Ups. It's trivial to filter transactions with a category of Unknown.

In addition, we have a superb new feature in 1.6.4 to allow the management of Categories (along with Payees) _directly_ during the Import process:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/c5d3f4ec-ac4a-4659-b390-da184ff4d21b)

I think we should let that feature bed down properly and get widespread use before changing something as fundamental as allowing categories not to be set or breaking people's workflow by adding Follow Up.

**Comment by vomikan** [comment]
> The `Unknown` category already gets applied.

Right. The `Unknown` is a category with some ID. You can find, rename, and delete this category. It exists.

I'm talking about the possibility of not setting a category in principle. I.e. the category ID will be -1 for a transaction.

**Comment by vomikan** [comment]
@tactilis You could test this DB. There is only one transaction without category. 
As you can see, it has a right to life. It is not reflected in the reports of Category, but it is reflected in the rest.
You will not be able to edit such a transaction, because the dialog will require you to specify a category.
My suggestion is to make the absence of a category valid if the statute is FollowUp.

Clear?


[test5924.zip](https://github.com/moneymanagerex/moneymanagerex/files/11929224/test5924.zip)

**Comment by tactilis** [comment]
> I'm talking about the possibility of not setting a category in principle. I.e. the category ID will be -1 for a transaction.

I understand that.

I'm trying to see what the benefit is compared to what we have. How would transactions with -1 be aggregated in reports for example?

If Category could be left blank, would you extend that approach to the Payee field as well?

And to the Amount?

I'm just asking whether it makes sense and whether it really brings a benefit.

**Comment by vomikan** [comment]
The status of trnsactions "Follow Up" does not matter much to us, because its functionality is not completed. This is visible because the transaction has a field in the database called followupId. And there is no table with a set of corresponding tasks referring to these IDs.
Later, I began to write the transaction color Id there (so as not to change the database structure).

Thus, the Follow Up status can be used as a temporary placement of unfilled transactions. They will not participate in statistics and reports. They can be used as templates and so on.
I think this is an interesting feature.

**Comment by ovari** [comment]
> It is not reflected in the reports of Category

Should it be reflected in the reports of Category with a [reserved word](https://en.wikipedia.org/wiki/Reserved_word) of the label being `Uncategorized` (where the translations are also reserved words)?

**Comment by tactilis** [comment]
@vomikan
> You could test this DB. There is only one transaction without category.
As you can see, it has a right to life. It is not reflected in the reports of Category, but it is reflected in the rest.
You will not be able to edit such a transaction, because the dialog will require you to specify a category.

But how did that transaction ever arise? Is seems like a bug to me.

**Comment by ovari** [comment]
> But how did that transaction ever arise? Is seems like a bug to me.

Importing from Intuit/Quicken software, where empty categories are allowed?

**Comment by vomikan** [comment]
> But how did that transaction ever arise? Is seems like a bug to me.

I've updated the transaction data outside the MMEX. It's not a bug, just huck for demo purposes.

**Comment by vomikan** [comment]
> Should it be reflected in the reports of Category with a [reserved word](https://en.wikipedia.org/wiki/Reserved_word) of the label being `Uncategorized` (where the translations are also reserved words)?

Yes, it shoild be. If we decide that transactions without a category are acceptable. This can't be happening at the moment.

**Comment by tactilis** [comment]
@vomikan 
> Right. The Unknown is a category with some ID. You can find, rename, and delete this category.

It doesn't matter. It is always recreated, if it is required, when Importing:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/19c800b8-a46c-428e-915d-054f3d29c1ac)

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/2f44b7ac-c503-4690-8436-fd13c4a81c47)

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/7e366157-890f-40da-90c2-22b523df557b)

**Comment by tactilis** [comment]
@ovari 
> Importing from Intuit/Quicken software, where empty categories are allowed?

See above.

**Comment by tactilis** [comment]
It seems to me that we already have something that is very simple and clear to understand.

In MMEX, all transactions have a category, even if it is the `Unknown` value. This makes it easy for the user to filter and report (and set Follow Up if they wish).

If you allow empty category then the filtering needs to change to cope.

We should be encouraging users to categorise their transactions, not leave them blank.

**Comment by vomikan** [comment]
> It is always recreated

We can redo it so that a category will not be created if it does not exist. This will simplify the import for data that does not have a category.
In reports, you can also display transactions without a category if desired (a small revision).

**Comment by ovari** [comment]
> If Category could be left blank, would you extend that approach to the Payee field as well?

Yes.

> And to the Amount?

Yes, to enable entering transactions without an amount to record interest rate changes.

```
Date: 12 June 2023
Payee: Credit Interest Rate
Note: 0.1%

Date: 14 June 2023
Payee: Debit Interest Rate
Note: 6.25%
```
 
> I'm just asking whether it makes sense and whether it really brings a benefit.

Separate reports with all payees that have `Credit Interest Rate` and `Debit Interest Rate` could be exported and used to check that the interest the financial institution is paying and charging is correct for the transactions using, say, [Statement Generator](https://extensions.libreoffice.org/en/extensions/show/21514).

**Comment by vomikan** [comment]
> Yes, to enable entering transactions without an amount to record interest rate changes.

This requires links from one transaction to another. And we don't have that.

**Comment by tactilis** [comment]
@ovari
> Yes, to enable **entering transactions** without an amount to record interest rate changes

This is manifestly not a transaction.

It is a note inserted into the transaction list.

If people want these, then they should be a special sort of Note, not bodged into a 'blank amount' transaction.

**Comment by ovari** [comment]
> It is always recreated

Creating data does not seem like a good idea as [round-trip format conversion](https://en.wikipedia.org/wiki/Round-trip_format_conversion) of the data would be broken.

**Comment by ovari** [comment]
> > Yes, to enable entering transactions without an amount to record interest rate changes.
> 
> This requires links from one transaction to another. And we don't have that.

There are no links.

Bank statements have transactions which include:
* debits
* credits
* credit rate changes

**Comment by ovari** [comment]
> It doesn't matter. It is always recreated, if it is required, when Importing

If someone uses one MMEX database file at different times in different languages, say English and Hungarian, is there a possibility that there will be two categories for `Uncategorized` transactions, one in English and another in Hungarian? If `-1` is used instead that allows for empty categories, would this situation not happen?

**Comment by vomikan** [comment]
> is there a possibility that there will be two categories for `Uncategorized` transactions, one in English and another in Hungarian?

No. Uncategorized will be wrapped in a special function _("Uncategorized"), as a result of which it will be displayed in the desired language.

**Comment by Lanna253** [comment]
That's crazy I never heard that an how do I begin to label a section needed when I don't know how I've always had issues with my funds acc it's like it never lets me update I've tried for the longest this not the 1st issue I had it's more frustrating then I'm giving light to I can try the link provided by normally what going to happened it's going to disappear on me or tell me I reverse revert something it's just these always happening just like so even when following the rules or guild I will put efforts but like I mentioned I have bad luck 🤞🤞🤞

**Comment by Lanna253** [comment]
It's has unseen unknowns ultrasound protection the system to cleaver an also recognized that I don't have or own a update up to date fund acc I try and it pop a redirect said no image these features come with facts it's will not 🚫 allow actions I've didn't mention recommendations or had put to action it's there own way to provide the source no unnecessarily action that not permitted why it labeled it's self uncategorized you see it's not that I'm being like this im telling the truth if anyone make up any input that is unseen examples it's a greed experiment 🥼🧪🥼 it's will never tell anyone this I had ridiculous experience it's watching everyone action, because I know this anything I try to do it's already place a stale link 🔗 anything I touch I get redirected to a dead link, it's the system own self management to ensure no one has authority aside from you it's a sign to as for myself I'm going to tell you what I said it's never allowing myself to have these behaviors I feel like it's testing who is going to be miss handling I have dumb experiences I can't not tell you how much I been effected by this it will never lets me have there behavior it will kick me out or make me time out or my acc going to pooft on me for day it means my acc ghosting me I have no access these are the system features not my own regardless how you approach me factor is the system know how long I had this acc and trust me it's not what you all see 🙈 greed experiment 🥼 they randomly do this often and I just am always like a test dummies I never had a fund acc it's just how it's happened these are problems I can't change and truth is they dissolve 🫠 they will make this go away they as in who? The money acc fact it's labeled that I'm shock you can't see what's going on it's a bait 🪤 fishing 🎣🦦🎣🦦🎣 it's odd don't you realize? I can't help it kick me out in a dead link if I try again band me for no reason please tell someone else I can't do a thing system already aware these no my actions it's watching who the leader is and I'm even understanding why I get these messages excuse me I tried and it kick me off

**Comment by luisdev** [comment]
Was any decision taken on this?
